### ISSUE TITLE
Claims missing from ClaimPrinciple when Authorizing a client by scope

### ISSUE THREAD

**Issue by agilenut** [IdentityServer/IdentityServer4#1365]
This seems like it should be easy but I'm having a difficult time figuring this out.

We have a service using IdentityServer4.AccessTokenValidation. 

Startup.cs
```
app.UseIdentityServerAuthentication(
    new IdentityServerAuthenticationOptions
    {
        Authority = "myIdentityServerUrl",
        RequireHttpsMetadata = false,
        ApiName = "myApi",
       ApiSecret = "myApiSecret"
    });
```

In that service, is an endpoint which needs to authorize based on the client's scope during a client credential flow. 

MyController.cs
```
[HttpPut]
[Route("{myId}")]
public async Task<IActionResult> Put(string myId, [FromBody] MyObject myObj)
{
    var policy = new AuthorizationPolicyBuilder().RequireAuthenticatedUser().RequireClaim("scope", "foo").Build();
    var isAuthorized = await _authorizationService.AuthorizeAsync(User, policy);

    if (!isAuthorized)
    {
        return new ChallengeResult();
    }
     ...
}
```

Unfortunately, User (the ClaimsPrinciple) never has any claims. So, the authorization always fails. 

I can see that the scope claim is available when I hit the introspection endpoint and it contains the scopes that I expect.

I can also see that the sub claim is available in the ClaimsPrinciple when I use a token obtained from an implicit grant flow. So, I know that some claims are being populated. 

I've tried playing around with the several of the properties on the Client and ApiResource but I'm obviously missing something. 

What do I need to do to have the scope claim added to the ClaimsPrinciple during a Client Crendentials flow?

**Comment by leastprivilege** [comment]
Check the logs on the API app - It seems token validation fails in some scenarios.

**Comment by agilenut** [comment]
@leastprivilege Thank you!

I looked at this again with fresh eyes this morning and found that my client credential test in Postman did not have "Bearer" in front of the access token. Once, I sent the correct authorization header, it worked as expected. 

I knew it had to be something trivial.  I looked at it longer than I'd like to admit. I suspected my client configuration but it was even more trivial!
